src/ansys/fluent/core/workflow.py

```diff
@@ -356,37 +356,43 @@ def python_name(self) -> str:
             Pythonic name of the task.
         """
         if not self._python_name:
-            display_name_map = self._command_source._help_string_display_text_map
-            if self.display_name() not in display_name_map.values():
-                try:
-                    this_command = self._command()
-                    # temp reuse helpString
-                    self._python_name = camel_to_snake_case(
-                        this_command.get_attr("helpString")
-                    )
-                    if (
-                        self._python_name
-                        in self._command_source._help_string_display_text_map
-                    ):
-                        self._populate_duplicate_task_list()
-                    else:
-                        self._command_source._help_string_display_text_map[
-                            self._python_name
-                        ] = self.display_name()
-                    self._command_source._help_string_command_id_map[
-                        self._python_name
-                    ] = this_command.command
-                    self._command_source._help_string_display_id_map[
-                        self._python_name
-                    ] = this_command.get_attr("displayText")
-                except Exception:
-                    pass
+            if self._command_source._dynamic_python_names:
+                display_name_map = self._command_source._python_name_display_text_map
+                if self.display_name() not in display_name_map.values():
+                    self._set_python_name()
+                else:
+                    self._python_name = list(display_name_map.keys())[
```

Comment:
```suggestion
                    self._python_name = list(display_name_map)[
```

Reply:
We might add a utility function and use it here

```
>>> def get_first_dict_key_for_value(d, value):
...   return next(key for key, val in d.items() if val == value)
...
>>> get_first_dict_key_for_value({1:2, 3:4, 4:4}, 4)
3
```

Reply:
@seanpearsonuk, for now shall we keep this locally and later move it to utils? Or, else we have to create a separate file in utils with this. Though I think the later to be a better idea as this can be generically used later as well. Please let me know your thoughts. Thank you.

Reply:
Done, Thank you.